ozark/src/main/java/org/glassfish/ozark/jersey/OzarkFeature.java

```diff
@@ -87,27 +101,47 @@ public void configure(FeatureContext context) {
         if (config.isRegistered(ViewResponseFilter.class)) {
             return;     // already registered!
         }
-        final boolean enableOzark = config.getClasses().stream().anyMatch(this::isController)
-                || config.getInstances().stream().map(o -> o.getClass()).anyMatch(this::isController);
-        if (enableOzark) {
-            context.register(ViewRequestFilter.class);
-            context.register(ViewResponseFilter.class);
-            context.register(ViewableWriter.class);
-            context.register(BindingInterceptorImpl.class);
-            context.register(OzarkModelProcessor.class);
-            context.register(CsrfValidateInterceptor.class);
-            context.register(CsrfProtectFilter.class);
-
-            // Initialize application config object in Mvc class
-            final BeanManager bm = CDI.current().getBeanManager();
-            final MvcContextImpl mvc = (MvcContextImpl) newBean(bm, MvcContextImpl.class);
-            mvc.setConfig(config);
-            mvc.setContextPath(servletContext.getContextPath());
+        Set<Class<?>> controllers = Stream.concat(config.getClasses().stream(), config.getInstances().stream().map(o -> o.getClass()))
+                .filter(this::isController).collect(Collectors.toSet());
+        if (controllers.isEmpty() || hasUnsupportedTypes(controllers)) {
```

Comment:
Yeah, the best way would actually be to fail with a runtime exception. But as you wrote in the comments this isn't possible as Jersey swallows all exceptions.

So it looks like we have two options:

1. Log an error message and abort the Ozark bootstrapping process. The application will continue to start up but all MVC functionality will simply not work. My fear is that people may overlook the error message and then wonder why nothing works as expected.
2. Log an warning message and continue to bootstrap Ozark. The application will continue to start up and MVC will basically work, but problematic controllers may fail with weird NPEs. Not so sure if this is much better than the first choice. In this case people may also overlook the warning.

I guess both ways are not very satisfying. We could do some research and try to find out why Jersey swallows the exception. Because this would be actually the best solution:

* Fail with an runtime exception which will break the application startup. The developer will immediately see the error and then knows how to fix it.